Acceptance tests for docker builder fail

Recently `ProvisionerHook` structure was changed (`ProvisionerTypes` field was added) but acceptance tests for docker builder were not adjusted accordingly. This results in:
```
PACKER_ACC=1 make test TEST=./builder/docker/

/go/src/github.com/hashicorp/packer/scripts/gofmtcheck.sh $(find . -not -path "./vendor/*" -name "*.go")
==> Checking that code complies with gofmt requirements...
Check passed.
2017/07/10 08:21:22 [DEBUG] Docker version: 17.4.0-ce
2017/07/10 08:21:22 [DEBUG] Container will be discarded
2017/07/10 08:21:22 ui: Creating a temporary directory for sharing data...
2017/07/10 08:21:22 Detected home directory from env var: /my/home/dir
2017/07/10 08:21:22 ui: Pulling Docker image: ubuntu
2017/07/10 08:21:22 Executing: /usr/bin/docker [pull ubuntu]
2017/07/10 08:21:22 ui: Using default tag: latest
2017/07/10 08:21:22 ui: latest: Pulling from library/ubuntu
2017/07/10 08:21:22 ui: Digest: sha256:a0ee7647e24c8494f1cf6b94f1a3cd127f423268293c25d924fbe18fd82db5a4
2017/07/10 08:21:22 ui: Status: Image is up to date for ubuntu:latest
2017/07/10 08:21:22 ui: Starting docker container...
2017/07/10 08:21:22 ui: Run command: docker run -v /my/home/dir/.packer.d/tmp/packer-docker647240145:/packer-files -d -i -t ubuntu /bin/sh
2017/07/10 08:21:22 Starting container with args: [run -v /my/home/dir/.packer.d/tmp/packer-docker647240145:/packer-files -d -i -t ubuntu /bin/sh]
2017/07/10 08:21:22 Waiting for container to finish starting
2017/07/10 08:21:22 ui: Container ID: aabccb2e1abd795215ef8f33a25289453b2269b1b3652a2d04874d750e92c457
2017/07/10 08:21:22 Running the provision hook
panic: runtime error: index out of range

goroutine 46 [running]:
github.com/hashicorp/packer/packer.(*ProvisionHook).Run(0xc42001dae0, 0xa37313, 0x10, 0xcef1a0, 0xc42001d040, 0x7fc0298fa238, 0xc4202622d0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/hashicorp/packer/packer/provisioner.go:66 +0x28a
github.com/hashicorp/packer/packer.(*DispatchHook).Run(0xc4201b0b70, 0xa37313, 0x10, 0xcef1a0, 0xc42001d040, 0x7fc0298fa238, 0xc4202622d0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/hashicorp/packer/packer/hook.go:68 +0x206
github.com/hashicorp/packer/common.(*StepProvision).Run.func1(0xc4201992c0, 0xcea360, 0xc4201b0b70, 0xcef1a0, 0xc42001d040, 0x7fc0298fa238, 0xc4202622d0)
        /go/src/github.com/hashicorp/packer/common/step_provision.go:39 +0x96
created by github.com/hashicorp/packer/common.(*StepProvision).Run
        /go/src/github.com/hashicorp/packer/common/step_provision.go:40 +0x235
FAIL    github.com/hashicorp/packer/builder/docker      0.721s
Makefile:77: recipe for target 'test' failed
make: *** [test] Error 1

```

--- a/builder/docker/communicator_test.go
+++ b/builder/docker/communicator_test.go
@@ -71,6 +71,7 @@ func TestUploadDownload(t *testing.T) {
 				upload,
 				download,
 			},
+			ProvisionerTypes: []string{"", ""},
 		},
 	}
 	hook := &packer.DispatchHook{Mapping: hooks}
@@ -161,6 +162,7 @@ func TestLargeDownload(t *testing.T) {
 				downloadCupcake,
 				downloadBigcake,
 			},
+			ProvisionerTypes: []string{"", "", ""},
 		},
 	}
 	hook := &packer.DispatchHook{Mapping: hooks}

